overlord/hookstate/ctlcmd/refresh.go

```diff
@@ -208,3 +209,49 @@ func (c *refreshCommand) printPendingInfo() error {
 	c.printf("%s", string(out))
 	return nil
 }
+
+func (c *refreshCommand) hold() error {
+	ctx := c.context()
+	ctx.Lock()
+	defer ctx.Unlock()
+	st := ctx.State()
+
+	// cache the action so that hook handler can implement default behavior
+	ctx.Cache("action", hookstate.HoldRefresh)
+
+	var affecting []string
+	if err := ctx.Get("affecting-snaps", &affecting); err != nil {
+		return fmt.Errorf("internal error: cannot get affecting-snaps")
+	}
+
+	// no duration specified, use maximum allowed for this gating snap.
+	var holdDuration time.Duration
+	if err := snapstate.HoldRefresh(st, ctx.InstanceName(), holdDuration, affecting...); err != nil {
+		// XXX: should we do something else here if we cannot hold anymore?
```

Comment:
as we discussed later we should likely let a snap hold again once for 1h, maybe we should leave a TODO for now